### PR TITLE
Add: Dropdown build menus from toolbar have icons

### PR DESCRIPTION
**Pull request by spnda**
Lately, the rail and road drop down menus have received icons for each entry in the menu.
I think, it looks strange if only 2 from 5 menus have these icons. And I wouldn't mind if every dropdown menu from the toolbar received these Icons, but that would require some new sprites.

### CODE REVIEW

File: src/toolbar_gui.cpp [hunk 1]
```diff
@@ -930,7 +930,11 @@ static CallBackFunction MenuClickBuildRoad(int index)
 
 static CallBackFunction ToolbarBuildWaterClick(Window *w)
 {
-	PopupMainToolbMenu(w, WID_TN_WATER, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, 1);
+	DropDownList *list = new DropDownList();
+
+	list->push_back(new DropDownListIconItem(SPR_IMG_BUILD_CANAL, PAL_NONE, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, WATER_CLASS_CANAL, false));
+
+	ShowDropDownList(w, list, 1, WID_TN_WATER, 140, true, true);
```
Comment (PeterN) [inline src/toolbar_gui.cpp]: Just use `0` as the value and default value here (and below) as there is only option and the value is not checked. Definitely wrong to use in the list item `WATER_CLASS_CANAL` and `1` as the default value.

File: src/toolbar_gui.cpp [hunk 2]
```diff
@@ -950,7 +954,11 @@ static CallBackFunction MenuClickBuildWater(int index)
 
 static CallBackFunction ToolbarBuildAirClick(Window *w)
 {
-	PopupMainToolbMenu(w, WID_TN_AIR, STR_AIRCRAFT_MENU_AIRPORT_CONSTRUCTION, 1);
+	DropDownList *list = new DropDownList();
+
+	list->push_back(new DropDownListIconItem(SPR_IMG_AIRPORT, PAL_NONE, STR_AIRCRAFT_MENU_AIRPORT_CONSTRUCTION, AIR_FAST, false));
+
+	ShowDropDownList(w, list, 2, WID_TN_AIR, 140, true, true);
```
Comment (PeterN) [inline src/toolbar_gui.cpp]: Same ^^

**Review by PeterN — DISMISSED** [review]

**Comment by PeterN** [comment]
Not really sure what this adds. The idea with rails and roads is it gives something to differentiate the different types, but with these menus we are not dealing with types.

File: src/toolbar_gui.cpp (hunk 1 above)
Comment (spnda) [reply src/toolbar_gui.cpp]: Which list item should I use instead?
Also, using 0 as the default value doesn't work in neither Waterways nor Airports. I have to use `1` or `2` for it to work. Is that related to the class?

Comment (PeterN) [reply src/toolbar_gui.cpp]: Set the value to 0 as well. No reason to use either of those enum values.

**Comment by spnda** [comment]
I do understand the original reason to add the icons to the menus. 
Though due to this the menus are not very consistent. Having these icons in other menus other than rail and road make the menus more consistent and it would also be a bit more filled, as it is quite empty just having text.

**Review by LordAro — DISMISSED** [review]
No harm in it, and a little consistency never went amiss

**Review by PeterN — DISMISSED** [review]
Does not build when rebased to master, needs the DropDownLists to be updated as in master.

File: src/toolbar_gui.cpp [hunk 3]
```diff
@@ -930,7 +930,11 @@ static CallBackFunction MenuClickBuildRoad(int index)
 
 static CallBackFunction ToolbarBuildWaterClick(Window *w)
 {
-	PopupMainToolbMenu(w, WID_TN_WATER, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, 1);
+	DropDownList *list = new DropDownList();
+
+	list->push_back(new DropDownListIconItem(SPR_IMG_BUILD_CANAL, PAL_NONE, STR_WATERWAYS_MENU_WATERWAYS_CONSTRUCTION, 0, false));
+
+	ShowDropDownList(w, list, 0, WID_TN_WATER, 140, true, true);
```
Comment (PeterN) [inline src/toolbar_gui.cpp]: `DropDownList list;`
`list.emplace_back(new ...);`
`ShowDropDownList(w, std::move(list), ...);`

**Review by PeterN — APPROVED** [review]

**Pull request merged**